homeassistant/components/remote_calendar/config_flow.py

```diff
@@ -0,0 +1,79 @@
+"""Config flow for Remote Calendar integration."""
+
+import logging
+from typing import Any
+
+from httpx import ConnectError, HTTPStatusError, UnsupportedProtocol
+from ical.calendar_stream import IcsCalendarStream
+from ical.exceptions import CalendarParseError
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_URL
+from homeassistant.helpers.httpx_client import get_async_client
+
+from .const import CONF_CALENDAR_NAME, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_CALENDAR_NAME): str,
+        vol.Required(CONF_URL): str,
+    }
+)
+
+
+class RemoteCalendarConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Remote Calendar."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+        errors: dict = {}
+        _LOGGER.debug("User input: %s", user_input)
+        self._async_abort_entries_match(
+            {CONF_CALENDAR_NAME: user_input[CONF_CALENDAR_NAME]}
+        )
+        self._async_abort_entries_match({CONF_URL: user_input[CONF_URL]})
+        client = get_async_client(self.hass)
+        try:
+            res = await client.get(user_input[CONF_URL], follow_redirects=True)
+            res.raise_for_status()
+        except UnsupportedProtocol as err:
```

Comment:
It seems like this set of errors from the httpx exception hierarchy feel a bit arbitrary, and its not catching all of them.
Having `HTTPError` as a catchall seems useful and `TimeoutException` seem common.
I don't see that `ValueError` is actually raised. 

I think I would drop `ConnectError` and `ValueError` and makde `HTTPStatusError and `HTTPError` raise `cannot_connect`

Same for the `coordinator.py`

Reply:
We can even drop HTTPStatusError, as it is a subclass of HTTPError?

Reply:
But we should keep `InvalidURL` as it is a different Class